### PR TITLE
Update PlayerTradeComponent.tsx

### PR DESCRIPTION
**Pull request by jimmyh77**
Added filtering to the player trade UI.  Options to filter Want and Offer trades.

### CODE REVIEW

File: src/scripts/ui/PlayerTradeComponent.tsx [hunk 1]
```diff
@@ -29,6 +125,9 @@ export function PlayerTradeComponent({ gameState, xy }: IBuildingComponentProps)
    const user = useUser();
    const myXy = getMyMapXy();
 
+   const [wantFilter, setWantFilter] = useState<keyof ResourceDefinitions|string>("");
```
Comment (insraq) [inline src/scripts/ui/PlayerTradeComponent.tsx:128]: Again, consider using a `Set`. If only single filter can be supported. the signature should be like `useState<Resource|null>`. Empty string is a bad item here to express null.

File: src/scripts/ui/PlayerTradeComponent.tsx [hunk 2]
```diff
@@ -60,7 +163,23 @@ export function PlayerTradeComponent({ gameState, xy }: IBuildingComponentProps)
                      <th>{t(L.PlayerTradeFrom)}</th>
                      <th></th>
                   </tr>
-                  {trades.map((trade) => {
+                  {trades
+                  .filter((t) => {
+                     if ( wantFilter === "" && offerFilter === "" ) {
```
Comment (insraq) [inline src/scripts/ui/PlayerTradeComponent.tsx:168]: This file probably needs to be reformatted.

**Review by insraq — CHANGES_REQUESTED** [review]

File: src/scripts/ui/PlayerTradeComponent.tsx (hunk 2 above)
Comment (jimmyh77) [reply src/scripts/ui/PlayerTradeComponent.tsx]: File has been formatted.

File: src/scripts/ui/PlayerTradeComponent.tsx (hunk 1 above)
Comment (jimmyh77) [reply src/scripts/ui/PlayerTradeComponent.tsx]: Simplified useStates and tidied up types.

**Comment by insraq** [comment]
Sorry for keeping this PR unmerged for so long - It seems it needs another rebase? I am also redoing the whole Player Trade UI. So it might be easier to close this for now?

**Comment by jimmyh77** [comment]
> Also, I am thinking maybe a generic sortable/filterable table view should be implemented.

I'm happy to wait until you've redesigned the layout.

And are you thinking something that can be used throughout the game in various places?

**Comment by insraq** [comment]
Unfortunately, I have to revert this PR as it causes significant stutters when changing filters:
image

**Pull request merged**